test/e2e/apps/deployment.go

```diff
@@ -898,19 +907,15 @@ func testProportionalScalingDeployment(f *framework.Framework) {
 	err = framework.WaitForReplicaSetDesiredReplicas(c.ExtensionsV1beta1(), secondRS)
 	Expect(err).NotTo(HaveOccurred())
 
-	numReplicaDiff := newReplicas - replicas
-	if numReplicaDiff < 0 {
-		numReplicaDiff = -numReplicaDiff
-	}
-	biggerProportion := (numReplicaDiff + 1) / 2
-	availableReplicas = availableReplicas + biggerProportion
-	framework.Logf("Verifying that first rollout's replicaset has %d available replicas", availableReplicas)
-	err = framework.WaitForReplicaSetAvailableReplicas(c.ExtensionsV1beta1(), firstRS, availableReplicas)
-	Expect(err).NotTo(HaveOccurred())
+	// First rollout's replicaset should have .spec.replicas = 8 + (30-10)*(8/13) = 8 + 12 = 20 replicas.
+	// Note that 12 comes from rounding (30-10)*(8/13) to nearest integer.
```

Comment:
@janetkuo Ignore my previous very long comment (I have deleted it in case you read it). During discussion by person, I mentioned about flooring the number, but I was wrong.

The mechanism, in fact, rounds the number to nearest integer. To clarify, in this example, `(30-10)*(8/13) = 12.31` and `(30-10)*(5/13) = 7.69`. By rounding to nearest integer, _12_ replicas are given to first rollout's RS, while _8_ replicas are given to second rollout's RS (supports test outcome). However, if we choose to floor `12.31` and `7.69`, 12 replicas are given to first rollout's RS, while _**7**_ replicas are given to second rollout's RS. The remaining _**1**_ leftover should be given to first rollout's RS because it has the most replicas, but this contradicts the test outcome.

I was confused yesterday by an edge case. Let's say there are 10 additional replicas and 6 active RSs having `1:1:1:1:1:1` proportion. Rounding to nearest integer makes all RSs to be given 2 replicas (a total of 12 replicas), but we only have 10 replicas. Flooring results in all RSs to be given 1 replica initially, with 4 leftovers. It looks like rounding to nearest integer does not make sense for this edge case. After further thinking, I realized that it is impossible to have 6 active RSs with same proportion during partial rollout (simply because it means all RSs have the same image with same `.spec.replicas`, hence no rollout occurs).